src/clients/HubPoolClient.ts

```diff
@@ -264,13 +264,15 @@ export class HubPoolClient extends BaseAbstractClient {
   l2TokenHasPoolRebalanceRoute(l2Token: Address, l2ChainId: number, hubPoolBlock = this.latestHeightSearched): boolean {
     return Object.values(this.l1TokensToDestinationTokensWithBlock).some((destinationTokenMapping) => {
       return Object.entries(destinationTokenMapping).some(([_l2ChainId, setPoolRebalanceRouteEvents]) => {
-        return setPoolRebalanceRouteEvents.some((e) => {
-          return (
-            e.blockNumber <= hubPoolBlock &&
-            e.l2Token.truncateToBytes20() === l2Token.truncateToBytes20() &&
-            Number(_l2ChainId) === l2ChainId
-          );
-        });
+        // sort events descending by block number
+        const sortedEvents = sortEventsDescending(setPoolRebalanceRouteEvents).filter(
```

Comment:
Is this right? This would return `false` if I, say, enabled token A at block 10, then disabled the route at block 20, then re-enabled at a different token, say token b, at block 30. If I query whether token b has a pool rebalance route, it does at block 31, but here it seems like we would return false since `sortedEvents[0].l2Token.truncateToBytes20() !== l2Token.truncateToBytes20()` since `tokenA !== tokenB`.